### DISCUSSION TITLE
How to check if a payload is attached with a Post call?

### DISCUSSION THREAD

**Discussion by maxtrunk** [honojs/hono#2451] [General]
Hi, I am new to HonoJs, and trying to figure out how to check if a post request has json payload attached. Hope someone can help me out. Thanks in advance. So here is my code:

```javascript
const factory = createFactory()

export const addClient = factory.createHandlers(async (c: Context) => {
	try {		
		const data =  c.req.json()
		
		return c.json({ "msg": "Success" }, 200)
	} catch (e) {
		return c.json({ "error": e ? e.toString() : "Unknown error on token debugging" }, 401)
	}
```

So, If I call my api via a POST request without a payload, then I am getting this error " SyntaxError: Unexpected end of JSON input"

I've tried the following solution, but none of them works:

```javascript
// Method 1:
	const data =  c.req.json()
	if(!data){
		return c.json({ "msg": "Vaild payload is required" }, 401)
	} // the if never triggered, because the error occurred in c.req.json()
// Method 2:
	const data = c.req.valid('json')
        // "data" always return "undefined" even I pass in json payload 
```

So how did u guy check the payload exists in Hono?

**Comment by NicoPlyley** [comment]
`c.req.json()` is a asynchronous you just need to await it. 

Hono will throw an error as well if there was no json sent, and you can customize the error on [`app.onError` ](https://hono.dev/api/hono#error-handling)

You will still need to check is an empty json object was sent.

**Reply by maxtrunk** [reply]
`app.onError` work!!! Thank you

**Reply by NicoPlyley** [reply]
Perfect, you can go ahead and close this out
